src/plugins/plugins.cpp

```diff
@@ -379,3 +379,54 @@ int drakvuf_plugins::start(const drakvuf_plugin_t plugin_id,
 
     return 0;
 }
+
+int drakvuf_plugins::stop(const drakvuf_plugin_t plugin_id)
+{
+    if ( __DRAKVUF_PLUGIN_LIST_MAX != 0 &&
+         plugin_id < __DRAKVUF_PLUGIN_LIST_MAX )
+    {
+        PRINT_DEBUG("Stopping plugin %s\n", drakvuf_plugin_names[plugin_id]);
+
+        if ( !drakvuf_plugin_os_support[plugin_id][this->os] )
+            return 0;
+
+        try
+        {
+            this->plugins[plugin_id]->stop();
+        }
+        catch (int e)
+        {
+            fprintf(stderr, "Plugin %s stop failed!\n", drakvuf_plugin_names[plugin_id]);
+            return -1;
+        }
+
+        PRINT_DEBUG("Stopping plugin %s finished\n", drakvuf_plugin_names[plugin_id]);
+        return 1;
+    }
+
+    return 0;
+}
+
+bool drakvuf_plugins::is_stopped(const drakvuf_plugin_t plugin_id)
+{
+    if ( __DRAKVUF_PLUGIN_LIST_MAX != 0 &&
+         plugin_id < __DRAKVUF_PLUGIN_LIST_MAX )
+    {
+        PRINT_DEBUG("Check is plugin %s stopped\n", drakvuf_plugin_names[plugin_id]);
+
+        if ( !drakvuf_plugin_os_support[plugin_id][this->os] )
+            return true;
+
+        try
+        {
+            return this->plugins[plugin_id]->is_stopped();
+        }
+        catch (int e)
+        {
+            fprintf(stderr, "Check is plugin %s stopped failed!\n", drakvuf_plugin_names[plugin_id]);
+            return true;
```

Comment:
In case this fails why is the default to return true?

Reply:
@tklengyel I'm pretty sure about default value `true`. Because this doesn't break current drakvuf logic.

Reply:
If you don't mind that we would change the value in the future.